packages/astro/src/core/app/node.ts

```diff
@@ -66,7 +66,11 @@ export class NodeApp extends App {
 		const hostname =
 			req.headers['x-forwarded-host'] ?? req.headers.host ?? req.headers[':authority'];
 		const port = req.headers['x-forwarded-port'];
-		const url = `${protocol}://${hostname}${port ? `:${port}` : ''}${req.url}`;
+
+		const portInHostname = typeof hostname === "string" && typeof port === "string" && hostname.endsWith(port);
+		const hostnamePort = portInHostname ? hostname : `${hostname}:${port}`;
```

Comment:
This should be ```
const hostnamePort = portInHostname ? hostname : `${hostname}${port ? `:${port}` : ''}`
``` because `port` can still be undefined.